vgaidarji/react-graphql-playground
# Set up ESLint + Prettier on the project

### What has been done
1. Configured `ESLint` https://eslint.org/ on the project
2. Configured `Prettier` https://prettier.io/ on the project
3. Fixed all codestyle issues (`npm run prettier-fix` script)
4. Included linting step on CI which runs both ESLint and Prettier 
5. Configured `Prettier ESLint` https://github.com/idahogurl/vs-code-prettier-eslint vscode extension to autoformat the codestyle issues on file save.

### How to test
1. Break any ESLint rule (remove `;`) and make sure `npm run lint` reports an issue. 
2. Run `npm run prettier-fix` or save the file in vscode and confirm that issue gets fixed
3. Re-run `npm run lint` and make sure no issues reported. 
4. Verify the same on CI
